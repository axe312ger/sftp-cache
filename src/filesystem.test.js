const { resolve } = require('path')
const fg = require('fast-glob')

const { getLocalFiles } = require('./filesystem')

jest.mock('fast-glob')

test('parses local directory', async () => {
  fg.mockImplementation(() => [
    {
      name: 'example-file-name.md',
      path: '/tmp/dir-to-cache/example-file-name.md',
      stats: { size: 4312, mtime: 1234 }
    },
    {
      name: 'file-in-root-dir.empty',
      path: '/tmp/dir-to-cache/file-in-root-dir.empty',
      stats: { size: 4312, mtime: 1234 }
    },
    {
      name: 'file-with-content.md',
      path: '/tmp/dir-to-cache/subdir/file-with-content.md',
      stats: { size: 4312, mtime: 1234 }
    }
  ])
  const root = resolve('/tmp')
  const dir = resolve(root, 'dir-to-cache')
  const tree = await getLocalFiles(dir, root)

  expect(tree).toMatchSnapshot()
})
