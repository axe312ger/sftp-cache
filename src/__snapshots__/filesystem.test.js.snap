// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses local directory 1`] = `
Array [
  Object {
    "name": "example-file-name.md",
    "path": "dir-to-cache/example-file-name.md",
    "stats": Object {
      "mtime": 1,
      "size": 4312,
    },
  },
  Object {
    "name": "file-in-root-dir.empty",
    "path": "dir-to-cache/file-in-root-dir.empty",
    "stats": Object {
      "mtime": 1,
      "size": 4312,
    },
  },
  Object {
    "name": "file-with-content.md",
    "path": "dir-to-cache/subdir/file-with-content.md",
    "stats": Object {
      "mtime": 1,
      "size": 4312,
    },
  },
]
`;
